auth_server/server/config.go

```diff
@@ -188,13 +188,18 @@ func validate(c *Config) error {
 			}
 			ghac.ClientSecret = strings.TrimSpace(string(contents))
 		}
-		if ghac.ClientId == "" || ghac.ClientSecret == "" || (ghac.TokenDB == "" && ghac.GCSTokenDB == nil) {
+		if ghac.ClientId == "" || ghac.ClientSecret == "" || (ghac.TokenDB == "" && (ghac.GCSTokenDB == nil && ghac.RedisTokenDB == nil)) {
 			return errors.New("github_auth.{client_id,client_secret,token_db} are required")
 		}
 
 		if ghac.ClientId == "" || ghac.ClientSecret == "" || (ghac.GCSTokenDB != nil && (ghac.GCSTokenDB.Bucket == "" || ghac.GCSTokenDB.ClientSecretFile == "")) {
 			return errors.New("github_auth.{client_id,client_secret,gcs_token_db{bucket,client_secret_file}} are required")
 		}
+
+		if ghac.ClientId == "" || ghac.ClientSecret == "" || (ghac.RedisTokenDB != nil && ghac.RedisTokenDB.NodeOptions.Addr == "" && len(ghac.RedisTokenDB.ClusterOptions.Addrs) < 1) {
```

Comment:
I left this change here but we should be able to do away with it. `go-redis` has a default value for `Addr` and `Addrs`. I kept it mainly because handling the configuration options will be weird if we don't set one of these explicitly:

For example:

```
redis_token_db:
    node_options:
    cluster_options:
```

The above would be a valid config but we won't know if we should create a single node connection or a cluster one. We can default to the single node one but it will be confusing to users.